libs/article-parser/src/lib/correlation-score/correlation-score.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ParsedArticleHead,
+  Parser,
+  ParsedArticle,
+  ParsedArticleParagraph,
+} from '@foodmedicine/interfaces';
+import * as fetch from 'node-fetch';
+import { correlationWeights, cutOffs } from './correlation-constants';
+import * as natural from 'natural';
+
+const tokenizer = new natural.WordTokenizer();
+
+async function downloadArticle(url: string): Promise<string> {
+  const ret = await fetch(url);
+  return await ret.text();
+}
+
+/**
+ * Find word frequencies through fuzzy search
+ * @param word
+ * @param paragraph
+ */
+function findWordFreqFuzzy(word: string, paragraph: string): number {
+  const tokenizedParagraph = tokenizer.tokenize(paragraph);
+  const overallFreqScore = tokenizedParagraph.reduce(
+    (freq: number, paragraphWord) => {
+      // distance ranges from 0 to 1. 1 being a perfect match
+      const distance = natural.JaroWinklerDistance(word, paragraphWord);
+      return freq + (distance > cutOffs.minimumWordDistance ? distance : 0);
+    },
+    0
+  );
+  return overallFreqScore;
+  // return paragraph.split(word).length - 1;
+}
+
+/**
+ * Compute the correlation score based off of the inputs
+ * Current features include impact frequencies, recommendation frequencies, impact x recommendation
+ * Paragraph length
+ * @param impactFreq
+ * @param recommendationFreq
+ */
+async function computeScore(
```

Comment:
Does this have to be async? Looks like everything is being done synchronously and you return a number instead of Promise<number>